src/Filename.php

```diff
@@ -53,6 +53,27 @@ public function exists() {
         return file_exists($this->name);
     }
 
+    /**
+     * @return bool
+     */
+    public function isWritable() {
+        if (is_writeable($this->asString())) {
+            return true;
+        }
+
+        if (is_link($this->asString())) {
+            // file_exists follows symlinks and returns false if the linked file does not exist
+            // so we need to check first if the file is a link
+            return false;
+        }
+
+        if (file_exists($this->asString())) {
+            return false;
```

Comment:
Why would the fact the file exists or not exists have anything to do with the question if it can be written to?

Reply:
Disregard this. If exists and would be writeable the 1st check would have caught this. So this check means it cannot be written to. So all is well here.